tests/aws/services/cloudformation/test_template_engine.py

```diff
@@ -243,6 +243,26 @@ def test_cfn_template_with_short_form_fn_sub(self, deploy_cfn_template):
         result = stack.outputs["Result"]
         assert result == "test"
 
+    @markers.aws.validated
+    def test_sub_number_type(self, deploy_cfn_template):
+        alarm_name_prefix = "alarm-test-latency-preemptive"
+        threshold = "1000.0"
+        period = "60"
+        stack = deploy_cfn_template(
+            template_path=os.path.join(
+                os.path.dirname(__file__), "../../templates/sub_number_type.yml"
+            ),
+            parameters={
+                "ResourceNamePrefix": alarm_name_prefix,
+                "RestLatencyPreemptiveAlarmThreshold": threshold,
+                "RestLatencyPreemptiveAlarmPeriod": period,
+            },
+        )
+
+        assert stack.outputs["AlarmName"] == alarm_name_prefix
```

Comment:
```suggestion
        assert stack.outputs["AlarmName"] == f"{alarm_name_prefix}-{period}"
```

Reply:
Thanks for the suggestion, I have addressed the comment in the latest commit.